tests/lib/rules/space-infix-ops.js

```diff
@@ -24,7 +24,9 @@ eslintTester.addRuleTest("lib/rules/space-infix-ops", {
         "(a + b) + (c + d)",
         "a = b",
         "a ? b : c",
-        "var a = b"
+        "var a = b",
+        { code: "a|0", args: [2, { int32Hint: true }] },
+        { code: "a |0", args: [2, { int32Hint: true }] }
```

Comment:
> I think this shouldn't be allowed, the hint should have no spaces before or after the |.

I see no reasons to remove only half of restrictions, described in #1295. Space before `|0` helps sometime to do code more readable, when left part is long. I use both `a|0` and `(something very long) |0` notations, depending on circumstances.

If someone needs more tuning with additional options - that can be arranged via separate issue & PR.